File: Python/Task24/n27686.py
# https://inf-ege.sdamgia.ru/problem?id=27686

file = open("n27421.txt", "r")
last = file.read(1)
len = maxLen = 0
while next := file.read(1): 
	if last == 'X' and next == 'X':
		len += 1
		maxLen = max(len, maxLen)
	else:
		len = 1
	last = next
print(maxLen)

File: C++/Task24/n27686/n27686.cpp
// https://inf-ege.sdamgia.ru/problem?id=27686

#include <iostream>
#include <fstream>

int main()
{
	std::fstream myfile(".\\Data\\24_demo.txt", std::ios_base::in);
	char c = 0;
	bool f = false;
	int len = 0, maxLen = 0;
	while (myfile >> c)
	{
		if (f == true)
		{
			if (c == 'X')
			{
				len++;
				maxLen = std::max(len, maxLen);
			}
			else // if (c != 'X')
			{
				f = false;
				len = 0;
			}
		}
		else // if (f == false)
		{
			if (c == 'X')
			{
				f = true;
				len = 1;
			}
		}
	}
	std::cout << maxLen << std::endl;
	return 0;
}